src/PagerView.tsx

```diff
@@ -144,19 +152,33 @@ export class PagerView extends React.Component<PagerViewProps> {
 
   render() {
     return (
-      <PagerViewView
-        {...this.props}
-        ref={(ref) => {
-          this.pagerView = ref;
+      <PagerViewContext.Provider
```

Comment:
[high]: 

React Context is a feature of React that allows you to pass data through the component tree without having to pass props down manually at every level. However, it is not a state management tool, and it has some performance implications that you should be aware of.

One of the main performance issues with React Context is that when a context value changes, all components that use the `useContext` hook or the `Consumer` component will re-render, regardless of whether they use the changed value or not. This can cause unnecessary re-rendering of components that are not affected by the context change, and thus affect the performance of your app.

[suggestion]
 Would you mind creating a separate file called `PagerViewContext` and moving the Context there? I don’t want to affect other projects. Moreover, this is breaking changes due to behavior changes.   After that, could you update the readme file with a section about the `usePager` hook?

Reply:
Thank you for your suggestion

When separating the React Context Provider into a separate file, there are issues with inserting the value. The fundamental problem is that re-rendering occurs even when the `usePager` hook is not used. (Originally, it was named `usePagerView`, but I changed it following your suggestion.)

I thought it would be good to solve the re-rendering issue without changing the way it is used. Therefore, I chose to use `useSyncExternalStore` to configure the store and then inject it into the context, changing the value of the store.

Since only the store’s value changes, rendering does not occur in components that do not use `usePager`.

However, one concern is that `useSyncExternalStore` is a React 18 feature, which is not compatible with earlier versions. But since the example also uses version 18, and it has been a while since version 18 was released, I believe it would be fine if we insert a warning message.

Lastly, one issue I have been contemplating is that even with the current usage, re-rendering occurs when there is a change in the page, even if one only wants to use setPage. For example, `const {setPage} = usePager();`

In `usePager()`, we could export only methods like `setPage`, `setPageWithoutAnimation`, `setScrollEnabled`, and construct a selector like `usePagerState((state) => state.page);`. This would perfectly optimize rendering only for the parts subscribing to the state, but it might make the usage somewhat more cumbersome.

If you agree with the approach I am taking, I would appreciate your opinion on this matter.

Reply:
In the `PagerView`, there's a `NonHookComponent` that does not use the usePager hook. Additionally, I have set it up to output a console.log statement every time it renders.

This code can be found in the example in the PR

* as-is render (only context)

https://github.com/callstack/react-native-pager-view/assets/41789633/e8e52e52-10b9-4db1-aa64-9ae4a46abab6

* to-be render (using `useSyncExternalStore`)

https://github.com/callstack/react-native-pager-view/assets/41789633/0435b56f-e1fe-478b-a8cb-2e29000ba1f5